plutus-core/plutus-core/src/PlutusCore/Builtin/Meaning.hs

```diff
@@ -209,7 +209,7 @@ class KnownMonotype val args res where
 
 -- | Once we've run out of term-level arguments, we return a
 -- 'TypeSchemeResult'/'RuntimeSchemeResult'.
-instance (Typeable res, KnownTypeAst (UniOf val) res, MakeKnown val res) =>
+instance (Typeable res, KnownTypeAst TyName (UniOf val) res, MakeKnown val res) =>
```

Comment:
why do we need to pin it down here?

Reply:
We want it here, because we want it in `TypeScheme`.

We want it in `TypeScheme`, because we hardcode `TyName` in 

```haskell
typeSchemeToType :: TypeScheme val args res -> Type TyName (UniOf val) ()
```

anyway, so there isn't much point in abstracting something to only be able to instantiate it to a single thing in the end. 

And the reason why it has to be a single thing is that otherwise we'd need to generalize

```
typeSchemeToType (TypeSchemeAll proxy schK) = case proxy of
    (_ :: Proxy '(text, uniq, kind)) ->
        let text = Text.pack $ symbolVal @text Proxy
            uniq = fromIntegral $ natVal @uniq Proxy
            a    = TyName $ Name text $ Unique uniq
        in TyForall () a (demoteKind $ knownKind @kind) $ typeSchemeToType schK
```

to be able to construct other `tyname`s and that would pretty much require redoing all name handling in `TypeScheme.hs`/`Meaning.hs`/`Elaborate.hs`, because that `uniq` in there assumes that we're going to construct a `TyName` in the end. Which would be a ton of work for nothing in return, because we can always just convert names.

As per my other comment, we really only benefit from generality in a single place (to the best of my knowledge), so why bother.